### ISSUE TITLE
Does not work in modern screenreaders/bad ARIA

### ISSUE THREAD

**Issue by aFarkas** [fnagel/jquery-ui#138]
Hi Felix,

The widget and the added ARIAs are wrong and it does not work right with any screenreader. The selectbox with a label and a value is transformed to menu-button with a name (label) but without a value. 

If I change the value using the cursor, the value visually changes, but there is no value, which changes. If I hit enter the popup doesn't really open.

I wrote an article, which describes how to create a proper select-menu which works in IE8+ and Firefox 3.5+ (You have to do some extras for IE8, because IE8 does not support aria-valuetext/aria-valuenow). You should really look into the [article (german)](http://www.protofunc.com/2009/09/23/wai-aria-widget-entwicklung-mit-accessibility-probeinspect-am-beispiel-einer-custom-select-drop-down-box/) and the [demo](http://www.protofunc.com/scripts/jquery/select/). 

If you want to discuss some parts, you can ping me. My skype name is diabolictrixta (Alexander Farkas Berlin).

cheers
alex

**Comment by fnagel** [comment]
Hey Alex, 

thanks for your contribution! The ARIA support was originally implemented by Scott from Filament Group a long time ago. I never checked it but I will take a look asap.
Im currently pretty busy and there is another major issue which needs to be fixed more urgent. Ive added this issue to milestone v1.3 (https://github.com/fnagel/jquery-ui/issues/milestones) which focus is on accessibility. I will contact you when I start working on this one!

Yours
Felix

ps: Ive learned a lot from your blog and mentioned some of it in my master thesis :-) Thanks a lot for that!

**Comment by fnagel** [comment]
Please take a look here as your knowledge could be very useful: https://github.com/fnagel/jquery-ui/issues/140

**Comment by fnagel** [comment]
Would you mind test the latest version? I will take a look at this one soon and contact you via skype!

**Comment by fnagel** [comment]
Alex and I skyped and decided to put our effort into the new version of selectmenu (see #140). 

Patches to improve this branch of selectmenu are still welcome and will be included!
